lib/pure/typetraits.nim

```diff
@@ -57,6 +57,16 @@ proc supportsCopyMem*(t: typedesc): bool {.magic: "TypeTrait".}
   ## This trait returns true iff the type ``t`` is safe to use for
   ## `copyMem`:idx:. Other languages name a type like these `blob`:idx:.
 
+proc sameType*(t1, t2: typedesc): bool =
```

Comment:
I predict the same thing with `name` will happen where people will ask for this to be renamed to `==`

Reply:
i thought about it... not sure which is better; sameType is more searchable but `==` is more predictable; Note: there's also `macros.sameType(Node,Node)` although obviously that one can't be named to `==`.
Overall `==` could be better ; should i go ahead and chage to `==`?

Reply:
done. replace with `==` ; which is nice because we get `int != float` for free.
makes the language nice and uniform

Reply:
This will quickly turn into ``compareTypes(a, b, flags)`` anyway, ``==`` is for simple comparisons, type equality is anything but simple. In fact, I know few other examples where it gets as complex as it gets for types.

Reply:
well nothing wrong w having both `==` for simple comparison and `compareTypes(a, b, flags)` to allow customizing what is meant by equality